RecoLocalCalo/EcalRecProducers/plugins/alpaka/AmplitudeComputationKernels.dev.cc

```diff
@@ -0,0 +1,326 @@
+                                  1e-11,
+                                  500,
+                                  16,
+                                  2);
+
+            calo::multifit::calculateChiSq(matrixL, pulse_matrix[idx], resultAmplitudes, samples[idx], chi2_now);
+
+            auto const deltachi2 = chi2_now - chi2;
+            chi2 = chi2_now;
+
+            if (std::abs(deltachi2) < 1e-3)
+              break;
+
+            //---- AM: TEST
+            //---- it was 3 lines above, now here as in the CPU version
+            ++iter;
+          }
+
+          // store to global output values
+          // FIXME: amplitudes are used in global directly
+          chi2s[inputCh] = chi2;
+          energies[inputCh] = resultAmplitudes(5);
+
+          CMS_UNROLL_LOOP
+          for (int counter = 0; counter < NPULSES; ++counter)
+            amplitudes[inputCh](counter) = resultAmplitudes(counter);
+        }
+      }
+    }
+  };
+
+  void minimization_procedure(Queue& queue,
+                              InputProduct const& digisDevEB,
+                              InputProduct const& digisDevEE,
+                              OutputProduct& uncalibRecHitsDevEB,
+                              OutputProduct& uncalibRecHitsDevEE,
+                              EventDataForScratchDevice& scratch,
+                              EcalMultifitConditionsDevice const& conditionsDev,
+                              ConfigurationParameters const& configParams,
+                              uint32_t const totalChannels) {
+    using DataType = SampleVector::Scalar;
+    // TODO: configure from python
+    auto threads_min = configParams.kernelMinimizeThreads[0];
+    auto blocks_min = cms::alpakatools::divide_up_by(totalChannels, threads_min);
+
+    auto workDivMinimize = cms::alpakatools::make_workdiv<Acc1D>(blocks_min, threads_min);
+    alpaka::exec<Acc1D>(queue,
+                        workDivMinimize,
+                        Kernel_minimize{},
+                        digisDevEB.const_view(),
+                        digisDevEE.const_view(),
+                        uncalibRecHitsDevEB.view(),
+                        uncalibRecHitsDevEE.view(),
+                        conditionsDev.const_view(),
+                        reinterpret_cast<::ecal::multifit::SampleMatrix*>(scratch.noisecovDevBuf.data()),
+                        reinterpret_cast<::ecal::multifit::PulseMatrixType*>(scratch.pulse_matrixDevBuf.data()),
+                        reinterpret_cast<::ecal::multifit::BXVectorType*>(scratch.activeBXsDevBuf.data()),
+                        reinterpret_cast<::ecal::multifit::SampleVector*>(scratch.samplesDevBuf.data()),
+                        scratch.hasSwitchToGain6DevBuf.data(),
+                        scratch.hasSwitchToGain1DevBuf.data(),
+                        scratch.isSaturatedDevBuf.data(),
+                        scratch.acStateDevBuf.data(),
+                        50);
```

Comment:
what is `50` ?

Reply:
Ah, it's the maximum number of iterations.
Would it make sense to make it configurable from the python configuration ?
Otherwise, can you just add a comment next to it, to explain what it is ?

Reply:
I will leave it hardcoded and add a comment. I do not think it has ever been changed since this is in production.